module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -0,0 +1,438 @@
+
+use Doctrine\ORM\EntityManager;
+use Laminas\Log\LoggerAwareInterface;
+use Laminas\Session\Container;
+use VuFind\Db\Entity\PluginManager as EntityPluginManager;
+use VuFind\Db\Entity\Resource;
+use VuFind\Db\Entity\User;
+use VuFind\Db\Entity\UserList;
+use VuFind\Db\Entity\UserResource;
+use VuFind\Exception\ListPermission as ListPermissionException;
+use VuFind\Exception\LoginRequired as LoginRequiredException;
+use VuFind\Exception\MissingField as MissingFieldException;
+use VuFind\Exception\RecordMissing as RecordMissingException;
+use VuFind\Log\LoggerAwareTrait;
+use VuFind\Tags;
+
+use function is_object;
+
+/**
+ * Database service for UserList.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+class UserListService extends AbstractService implements LoggerAwareInterface, ServiceAwareInterface
+{
+    use LoggerAwareTrait;
+    use ServiceAwareTrait;
+
+    /**
+     * Tag parser.
+     *
+     * @var Tags
+     */
+    protected $tagParser;
+
+    /**
+     * Session container for last list information.
+     *
+     * @var Container
+     */
+    protected $session = null;
+
+    /**
+     * Constructor
+     *
+     * @param EntityManager       $entityManager       Doctrine ORM entity manager
+     * @param EntityPluginManager $entityPluginManager VuFind entity plugin manager
+     * @param Tags                $tagParser           Tag parser
+     * @param Container           $session             Session container
+     */
+    public function __construct(
+        EntityManager $entityManager,
+        EntityPluginManager $entityPluginManager,
+        Tags $tagParser,
+        Container $session = null
+    ) {
+        parent::__construct($entityManager, $entityPluginManager);
+        $this->tagParser = $tagParser;
+        $this->session = $session;
+    }
+
+    /**
+     * Get an array of resource tags associated with the list.
+     *
+     * @param UserList $list UserList object.
+     *
+     * @return array
+     */
+    public function getResourceTags($list)
+    {
+        $user = $list->getUser();
+        $tags = $this->getDbService(\VuFind\Db\Service\TagService::class)
+            ->getListTagsForUser($user, null, $list);
+        return $tags;
+    }
+
+    /**
+     * Create a userlist entity object.
+     *
+     * @return UserList
+     */
+    public function createUserList(): UserList
+    {
+        $class = $this->getEntityClass(UserList::class);
+        return new $class();
+    }
+
+    /**
+     * Create a new list object.
+     *
+     * @param User|bool $user User object representing owner of
+     * new list (or false if not logged in)
+     *
+     * @return UserList|bool
+     * @throws LoginRequiredException
+     */
+    public function getNew($user)
```

Comment:
There is a [UserList unit test](https://github.com/vufind-org/vufind/blob/dev/module/VuFind/tests/unit-tests/src/VuFindTest/Db/Table/UserListTest.php) that covers the old version of getNew. It might be a good idea to start a UserListServiceTest and port those existing tests over so we don't reduce our code coverage when we remove the old code.